doc/dev/adr/0026-wot-protocol-binding.md

```diff
@@ -0,0 +1,355 @@
+# ADR 26: WoT Protocol Binding for AIO
+
+## Context
+
+The AIO team is planning to change our primary modeling language from [Digital Twins Definition Language (DTDL)](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v4/DTDL.v4.md) to [Web of Things (WoT) Thing Description (TD)](https://www.w3.org/TR/wot-thing-description/).
+The AIO ProtocolCompiler currently ingests DTDL models that employ extension types and properties defined by the [DTDL Mqtt extension](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v4/DTDL.mqtt.v3.md).
+This extension enables models to express MQTT topic patterns, header value definitions, error annotations, and other information that is not included in core DTDL but which is needed for generating source-code binders that target the AIO SDKs.
+
+Models written in WoT TD format will need to express the same information that is currently expressed via DTDL and its Mqtt extension.
+WoT TD can directly express much of this information, but some things &mdash; such as MQTT topic patterns &mdash; have no native representation in WoT.
```

Comment:
Please check the MQTT protocol binding if it supports it: https://w3c.github.io/wot-binding-templates/bindings/protocols/mqtt/